### PR TITLE
fix: plural of date picker

### PR DESCRIPTION
**Pull request by louisphn**
- [x] fix plural of datepicker (Last 1 hours -> Last 1 hour, Last 1 years -> Last 1 year)

Please take a look at the preview screenshots and let me know if there is other issues.
Thanks!

image
image

### CODE REVIEW

**Comment by louisphn** [comment]
By the way because the CLA did not get updated so can I make use of this instead?
image

**Comment by petethepig** [comment]
@louisphn  Thanks for the contribution!

I think there's just an issue connecting your github account to the email. You need to either add the email you used to sign the CLA to your github account or sign it again with an email that is attached to your github account.

File: webapp/javascript/util/formatDate.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ export function readableRange(from: string, until: string) {
   const dateFormat = 'yyyy-MM-dd hh:mm a';
   if (/^now-/.test(from) && until === 'now') {
     const { number, _multiplier } = convertPresetsToDate(from);
-    return `Last ${number} ${_multiplier}`;
+    const multiplier =
+      parseInt(number, 10) >= 2 ? _multiplier : _multiplier.slice(0, -1);
```
Comment (eh-am) [inline webapp/javascript/util/formatDate.ts:48]: I don't quite understand what's going on here, can you explain?

**Review by eh-am — COMMENTED** [review]
Awesome, thank you so much for your contribution!

The only thing I would ask is if you can figure out why this test was NOT failing.
https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/util/formatDate.spec.ts#L10

**Comment by louisphn** [comment]
@eh-am looks like the 'readableRange' test was skipped so it was not failing. When I changed `test.each` to `test.only.each` the test now works. Should I be pushing this to the PR?

image
image

Because it is `test.only`, all the other tests in file are skipped, except for the one with `only` like `now-1h` in `formatAsObject` test
image

**Comment by eh-am** [comment]
@louisphn ahh that makes sense!

I think it's easier to just remove this `.only` instead?  https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/util/formatDate.spec.ts#L48

**Comment by louisphn** [comment]
@eh-am true haha should have done that instead of adding only to each test
I fixed that in the latest commit so please take a look when you have time. (the test passed by the way)
Thanks!

**Comment by eh-am** [comment]
Almost there, the tests are still failing but that's probably my fault. Here's what you should do:

```diff
diff --git a/webapp/javascript/util/formatDate.spec.ts b/webapp/javascript/util/formatDate.spec.ts
index 3bfab904..f00dfb91 100644
--- a/webapp/javascript/util/formatDate.spec.ts
+++ b/webapp/javascript/util/formatDate.spec.ts
@@ -42,7 +42,7 @@ describe('FormatDate', () => {
     it('works with "now"', () => {
       // TODO
       // not entirely sure this case is even possible to happen in the code
-      expect(formatAsOBject('now')).toEqual(mockDate);
+      expect(formatAsOBject('now')).toEqual(mockDate.getTime());
     });
 
     it('works with "now-1h"', () => {
@@ -58,12 +58,12 @@ describe('FormatDate', () => {
     });
 
     it('works with "now-1m"', () => {
-      expect(formatAsOBject('now-1m')).toBe(1640090641741);
+      expect(formatAsOBject('now-1m')).toBe(1640090581741);
     });
 
     it('works with absolute timestamps', () => {
       expect(formatAsOBject('1624192489')).toEqual(
-        new Date('2021-06-20T12:34:49.000Z')
+        new Date('2021-06-20T12:34:49.000Z').getTime()
       );
     });
   });

````

Also it looks like you haven't signed yet, can you have a look at it? 
https://github.com/pyroscope-io/pyroscope/pull/831#issuecomment-1038687089

Thank you again @louisphn!

**Comment by louisphn** [comment]
@eh-am I have fixed according to your code. Please take a look when you have time. Thanks!
I have signed the CLA but the first commit was not linked to my email address at the time I pushed so it shows there are 2 committers which I don't know how to sign on behalf of the other username. Do you have any idea how to fix this?

**Comment by eh-am** [comment]
@petethepig can you shed some light RE the license thing?

**Pull request merged**